### PR TITLE
Trigger onValidate when inital value is set on Select widget

### LINKED ISSUE #1653: Select isn't validating initial value
`Select` is reporting false validity when using an initial value.

https://codesandbox.io/s/distracted-meadow-2tdj1?file=/src/main.tsx

Click the select box to open it and then closing it will cause validity to report true.

### PR DESCRIPTION
**Pull request by samends**
**Type:** bug

**Description:**
`Select` widget was not validating when there was an initial value set. Pending unit test

Resolves #1653

### CODE REVIEW

File: src/select/index.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ export const Select = factory(function Select({
 		data
 	} = get(options(), { read, meta: true });
 
-	if (required && dirty) {
+	if ((required && dirty) || value !== undefined) {
```
Comment (aciccarello) [inline src/select/index.tsx:142]: The extra parenthesis shouldn't be necessary
```suggestion
	if (required && dirty || value !== undefined) {
```

**Review by aciccarello — APPROVED** [review]

**Review by tomdye — CHANGES_REQUESTED** [review]
I'm not sure this is right as its validating selects even when they are not required and do not require validation.

**Comment by samends** [comment]
@tomdye this was added so that if there's an initial value set, the `Select` widget will validate it

**Comment by tomdye** [comment]
@samends I believe that any passed `onValidate` function should be called but if `required` is not set to true it should not validate internally.

**Review by tomdye — APPROVED** [review]

**Pull request merged**